src/orders/components/OrderDraftDetailsSummary/OrderDraftDetailsSummary.tsx

```diff
@@ -156,9 +156,9 @@ const OrderDraftDetailsSummary: React.FC<OrderDraftDetailsSummaryProps> = props
           <td>
             <Popover open={isDialogOpen}>
               <Popover.Trigger>
-                <Button variant="tertiary" onClick={isDialogOpen ? closeDialog : openDialog}>
+                <Box onClick={isDialogOpen ? closeDialog : openDialog}>
```

Comment:
Issue: This is not accessible and you can't select it with a keyboard. 

I would propose a Link variant to the Button, something like this:
```
                <Button
                  variant="tertiary"
                  onClick={isDialogOpen ? closeDialog : openDialog}
                  __backgroundColor="transparent"
                  textDecoration={{
                    hover: "underline",
                    focus: "underline",
                  }}
                  style={{
                    textUnderlineOffset: "4px",
                    padding: 0,
                    margin: 0,
                    height: "1rem",
                    textDecorationColor: vars.colors.border.accent1,
                  }}
                >
                  <Link>{intl.formatMessage(discountTitle)}</Link>
                </Button>
                ```